pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -296,34 +296,40 @@ def __init__(
             if output_type is not str:
                 raise TypeError('`result_type` and `output_type` cannot be set at the same time.')
             warnings.warn('`result_type` is deprecated, use `output_type` instead', DeprecationWarning)
-            output_type = _deprecated_kwargs['result_type']
+            output_type = _deprecated_kwargs.pop('result_type')
 
         self.output_type = output_type
 
         self.instrument = instrument
 
         self._deps_type = deps_type
 
-        self._deprecated_result_tool_name = _deprecated_kwargs.get('result_tool_name')
+        self._deprecated_result_tool_name = _deprecated_kwargs.pop('result_tool_name', None)
         if self._deprecated_result_tool_name is not None:  # pragma: no cover
             warnings.warn(
                 '`result_tool_name` is deprecated, use `output_type` with `ToolOutput` instead',
                 DeprecationWarning,
             )
 
-        self._deprecated_result_tool_description = _deprecated_kwargs.get('result_tool_description')
+        self._deprecated_result_tool_description = _deprecated_kwargs.pop('result_tool_description', None)
         if self._deprecated_result_tool_description is not None:  # pragma: no cover
             warnings.warn(
                 '`result_tool_description` is deprecated, use `output_type` with `ToolOutput` instead',
                 DeprecationWarning,
             )
-        result_retries = _deprecated_kwargs.get('result_retries')
+        result_retries = _deprecated_kwargs.pop('result_retries', None)
         if result_retries is not None:  # pragma: no cover
             if output_retries is not None:
                 raise TypeError('`output_retries` and `result_retries` cannot be set at the same time.')
             warnings.warn('`result_retries` is deprecated, use `max_result_retries` instead', DeprecationWarning)
             output_retries = result_retries
 
+        if _deprecated_kwargs:
+            from .exceptions import UserError
+
+            unknown_kwargs = ', '.join(f'`{k}`' for k in _deprecated_kwargs.keys())
+            raise UserError(f'Unknown keyword arguments: {unknown_kwargs}')
```

Comment:
Can you please move this to a util function to save some duplication? That'll also make it easier to get test coverage!